### PR TITLE
py-pyqt5: add v5.15.9

### PR DESCRIPTION
**Pull request by adamjstewart**
Extracted from #32696

Can't test because of #34064

Closes #29039 @manuelakuhn 
Closes #21554 @mwm126

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyqt5/package.py [hunk 1]
```diff
@@ -12,26 +12,41 @@ class PyPyqt5(SIPPackage):
     Windows, OS X, Linux, iOS and Android. PyQt5 supports Qt v5."""
 
     homepage = "https://www.riverbankcomputing.com/software/pyqt/intro"
-    url = (
-        "https://www.riverbankcomputing.com/static/Downloads/PyQt5/5.13.0/PyQt5_gpl-5.13.0.tar.gz"
-    )
-    list_url = "https://www.riverbankcomputing.com/software/pyqt/download5"
+    url = "https://files.pythonhosted.org/packages/source/P/PyQt5/PyQt5-5.15.9.tar.gz"
+    list_url = "https://pypi.org/simple/PyQt5/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyqt5/package.py:16]: URL no longer works and older versions are no longer available for download so I deprecated them

**Comment by Sinan81** [comment]
I'm on vacation. Will look into this in a week.

**Comment by manuelakuhn** [comment]
Just to give some feedback in the meantime while waiting for @Sinan81:

`py-sip`: The `py-toml` dependency is only needed for `python@:10` and is listed in `install_requires`, shouldn't it be a run dependency as well? (currently only build dependency)

`py-pyqt-builder`, `py-sip` (even without py-toml as run dependency) and `py-pyqt5-sip` install fine for me with `--test=root` but unfortunately `py-pyqt5@5.15.9` does and throws the following error:
```
==> py-pyqt5: Executing phase: 'configure'
==> [2023-07-06-20:15:26.181278] '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/python-3.10.12-kc4fec4hcpsvkjtrhawqif564lshi7k3/bin/python3.10' 'configure.py' '--pyuic5-interpreter' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/python-3.10.12-kc4fec4hcpsvkjtrhawqif564lshi7k3/bin/python3.10' '--sipdir' '/home/nela/projects/manuelakuhn/spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/share/sip/PyQt5' '--designer-plugindir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/plugins/designer' '--qml-plugindir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/plugins/PyQt5' '--stubsdir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/lib/python3.10/site-packages/PyQt5' '--verbose' '--confirm-license' '--qmake' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/qt-5.15.10-5tcv6rxm4pl57fwaqzduol6zl6h7gw4o/bin/qmake' '--sip' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-sip-6.7.9-h3wsyjqlg7ifh65hohiddw22f3tsuvcy/bin/sip' '--sip-incdir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-sip-6.7.9-h3wsyjqlg7ifh65hohiddw22f3tsuvcy/include/python3.10' '--bindir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/bin' '--destdir' '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/lib/python3.10/site-packages'
$spack-stage/spack-stage-py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/spack-src/configure.py:21: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils import sysconfig
$spack-stage/spack-stage-py-pyqt5-5.15.9-exmbv6feat4bxzt3hmbypuzdeb4yfa5k/spack-src/configure.py:21: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
  from distutils import sysconfig
Usage: python configure.py [opts] [name=value] [name+=value]

configure.py: error: '$spack/opt/spack/linux-fedora37-x86_64/gcc-12.3.1/py-sip-6.7.9-h3wsyjqlg7ifh65hohiddw22f3tsuvcy/bin/sip' is not an executable
```
Same error happens for `py-pyqt5@5.13.1` where the old version (before this PR) it installs fine.

**Comment by adamjstewart** [comment]
Might give up on the deprecated `configure.py` build and use `sip-build` instead like I did in #32696. I know that worked once upon a time. Unfortunately that seemed to break some of the packages that depend on it, although matplotlib (the only thing I care about) worked. Let me see if I can find somewhere to actually test this, might have access to a Linux machine.

**Comment by adamjstewart** [comment]
Sub-seded by #32696. Going to hack on all of them together again since this requires changes to the base class.

**Pull request closed without merging**